iqrs_method.py

```diff
@@ -89,18 +79,32 @@ def lambda_handler(event, context):
             return final_output
 
 
-def calc_iqrs(input_table, move_cols, iqrs_cols):
-    distinct_strata_region = input_table[['region', 'strata']].drop_duplicates()
+def calc_iqrs(input_table, move_cols, iqrs_cols, distinct_values):
+    """
+    Calculate IQRS.
+    :param input_table: Input DataFrame. - Type: DataFrame
+    :param move_cols: Movement column list. - Type: List
+    :param iqrs_cols: IQRS column list. - Type: List
+    :param distinct_values: Array of column names to derive distinct values from
+                            and store in table. - Type: List
+    :return: Table. - Type: DataFrame
+    """
+    distinct_strata_region = input_table[distinct_values].drop_duplicates()
+    iqr_filter = ""
+
+    for value in distinct_values:
+        if value != distinct_values[0]:
+            iqr_filter += " & "
+        iqr_filter += "(input_table[\"%s\"] == row[%s])"\
+            % (value, distinct_values.index(value))
+
     for row in distinct_strata_region.values:
-        iqr_filter = (input_table["region"] == row[0]) & (input_table["strata"] == row[1])  # noqa: E501
-        filtered_iqr = input_table[iqr_filter]
-        # Pass the question number and region and strata groupping to the
-        # iqr_sum function.
+        filtered_iqr = input_table[pd.eval(iqr_filter)]
```

Comment:
I was under the impression we didn't want to be using eval? Ian had asked Tom to take eval out completely so he may want us to here.